src/commands/epgsql_cmd_copy_done.erl

```diff
@@ -0,0 +1,51 @@
+%%% @doc Tells server that the transfer of COPY data is done.
+%%%
+%%% It makes server to "commit" the data to the table and switch to the normal command-processing
+%%% mode.
+%%%
+%%% @see epgsql_cmd_copy_from_stdin
+
+-module(epgsql_cmd_copy_done).
+-behaviour(epgsql_command).
+-export([init/1, execute/2, handle_message/4]).
+-export_type([response/0]).
+
+-type response() :: {ok, Count :: non_neg_integer()}
+                  | {error, epgsql:query_error()}.
+
+%% -include("epgsql.hrl").
+-include("protocol.hrl").
+-include("../epgsql_copy.hrl").
+
+init(_) ->
+    [].
+
+execute(Sock0, St) ->
+    #copy{format = Format} = epgsql_sock:get_subproto_state(Sock0), % assert we are in copy-mode
+    Sock1 = epgsql_sock:set_packet_handler(on_message, Sock0),
+    Sock = epgsql_sock:set_attr(subproto_state, undefined, Sock1),
+    {PktType, PktData} = epgsql_wire:encode_copy_done(),
+    case Format of
+        text ->
+            {send, PktType, PktData, Sock, St};
+        binary ->
+            Pkts = [{?COPY_DATA, epgsql_wire:encode_copy_trailer()},
+                    {PktType, PktData}],
+            {send_multi, Pkts, Sock, St}
+    end.
+
+handle_message(?COMMAND_COMPLETE, Bin, Sock, St) ->
+    Complete = epgsql_wire:decode_complete(Bin),
+    Res = case Complete of
+        {copy, Count} -> {ok, Count};
+        copy -> ok
```

Comment:
This seems like it's never a possibility from `epgsql_wire:decode_complete`, nor it would fulfill the `response()` type above.

Reply:
Yeah, it is an artifact from me changing line 220 in epgsql_wire much later after I added this snippet. Seems it should be relatively safe to assume it always returns `COPY 1234` and never just `COPY`:

> CommandComplete (B)
> <..>
> For a COPY command, the tag is COPY rows where rows is the number of rows copied. (Note: the row count appears only in PostgreSQL 8.2 and later.)

https://www.postgresql.org/docs/current/protocol-message-formats.html

> On successful completion, a COPY command returns a command tag of the form
> 
> COPY count
> The count is the number of rows copied.

https://www.postgresql.org/docs/current/sql-copy.html#id-1.9.3.55.7

So I think it's safe as long as we don't support PostgreSQL older than 8.2 (postgresql.org says the oldest supported postgres server version is 9.5; but I think I heard that AWS Redshift uses older version?)